### PR TITLE
Update setup_dependencies.org

### PR DESCRIPTION
**Pull request by rmahique**
- Added a section with the instructions to generate the necessary files mentioned and used in the exercises.
- Added missing dependencies

### CODE REVIEW

File: setup_dependencies.org [hunk 1]
```diff
@@ -33,8 +33,16 @@ https://github.com/libbpf/libbpf.
...
 ** libxdp and libbpf as git-submodules
 
-This repository uses both [[https://github.com/xdp-project/xdp-tools/][libxdp]] and [[https://github.com/libbpf/libbpf][libbpf]] as git-submodules. After cloning
-this repository you need to run the command:
+This repository uses both [[https://github.com/xdp-project/xdp-tools/][libxdp]] and [[https://github.com/libbpf/libbpf][libbpf]] as git-submodules.
+
+We will start by cloning this repository:
+
+#+begin_example
+git clone git@github.com:xdp-project/xdp-tutorial.git
```
Comment (tohojo) [inline setup_dependencies.org:41]: We use the https URL everywhere else; let's do the same here...

File: setup_dependencies.org [hunk 2]
```diff
@@ -91,7 +99,7 @@ Note that you need to do this in the shell you are using to load programs
 On Debian and Ubuntu installations, install the dependencies like this:
 
 #+begin_example
- $ sudo apt install clang llvm libelf-dev libpcap-dev build-essential
+ $ sudo apt install clang llvm libelf-dev libpcap-dev build-essential libc6-dev-i386
```
Comment (tohojo) [inline setup_dependencies.org:102]: Why is this needed?

File: setup_dependencies.org [hunk 3]
```diff
@@ -203,3 +211,25 @@ On a machine running the openSUSE Tumbleweed distribution, install package:
...
+#+begin_example
+ $ ./configure
+#+end_example
+
+If there is a missing dependency it should output some error, if not we can continue.
+
+Now run make to generate all the necessary files:
+
+#+begin_example
+ $ sudo make
```
Comment (tohojo) [inline setup_dependencies.org:231]: Absolutely do not run `make` with sudo!

File: setup_dependencies.org (hunk 1 above)
Comment (tohojo) [reply setup_dependencies.org:41]: Also, if we're adding this, I think it's better to add the `--recurse-submodules` switch here (and get rid of it below), then rewrite the text below to explain that the `git submodule update` command is only needed if `--recurse-submodules` was not used in the initial clone command...

**Review by tohojo — CHANGES_REQUESTED** [review]
Thank you for the additions! A few comments below, but overall looks like a nice update :)

File: setup_dependencies.org (hunk 2 above)
Comment (tohojo) [reply setup_dependencies.org:96]: Right, but I mean, what breaks if it's not present?

Comment (tohojo) [reply setup_dependencies.org:96]: Huh, didn't even realise that was in there. Alright, cool :)

Comment (sys0613) [reply setup_dependencies.org:96]: how to deal this

clang: 13.0.1 (UnionTech OS 13.0.1-2.0.2.module+uelc20+911+09ad4a12.01)
libbpf support: submodule
ELF support: yes
zlib support: yes
libxdp support: submodule
Configuring libxdp to use our libbpf submodule
Found clang binary 'clang' with version 13 (from 'clang version 13.0.1 (UnionTech OS 13.0.1-2.0.2.module+uelc20+911+09ad4a12.01)')
not using emacs:
using bpftool v7.5.0
libbpf support: custom v1.2.0
  perf_buffer__consume support: yes
  btf__load_from_kernel_by_id support: yes
  btf__type_cnt support: yes
  bpf_object__next_map support: yes
  bpf_object__next_program support: yes
  bpf_program__insn_cnt support: yes
  bpf_program__type support: yes
  bpf_program__flags support: yes
  bpf_program__expected_attach_type support: yes
  bpf_map_create support: yes
  perf_buffer__new_raw support: yes
  bpf_xdp_attach support: yes
  bpf_map__set_autocreate support: yes
  bpf_prog_test_run_opts support: yes
  bpf_xdp_query support: yes
zlib support: yes
ELF support: yes
pcap support: yes
secure_getenv support: yes
cap-ng support: no
*** ERROR - Clang BPF-prog cannot include <errno.h>
          - Install missing userspace header file

Compile error: In file included from config.0WzjSh/bpf_use_errno_test.c:1:
In file included from /usr/include/errno.h:25:
In file included from /usr/include/features.h:462:
/usr/include/gnu/stubs.h:7:11: fatal error: 'gnu/stubs-32.h' file not found
# include <gnu/stubs-32.h>
          ^~~~~~~~~~~~~~~~
1 error generated.

 On Fedora install:
   dnf install glibc-devel.i686
 On Debian install:
   apt install libc6-dev-i386

**Pull request merged**